sdks/go/pkg/beam/artifact/materialize.go

```diff
@@ -150,6 +158,12 @@ func extractStagingToPath(artifact *pipepb.ArtifactInformation) (string, error)
 			return "", err
 		}
 		stagedName = generateId() + "-" + filepath.Base(ty.Path)
+	} else if artifact.TypeUrn == URNUrlArtifact {
```

Comment:
I imaging the mixing of requesting TypeUrn and RoleUrn in this if/else ladder is an artifact of legacy handling?

Reply:
If the staging_to role is defined, we follow the naming definition from the role payload. However we just generate the staging name when the file name is not important. For example, the name is not important for Python pip requirement file so we use pip_requirement role instead of staging_to role.